config/initializers/secure_headers.rb

```diff
@@ -10,8 +10,19 @@
                      IdentityConfig.store.component_previews_enabled
 
   config.action_dispatch.default_headers.merge!(
-    'X-Frame-Options' => previews_enabled ? 'SAMEORIGIN' : 'DENY',
     'X-XSS-Protection' => '1; mode=block',
     'X-Download-Options' => 'noopen',
   )
+
+  if Identity::Hostdata.env == 'dev'
+    # so we can embed a special lookbook component into the dev docs
+    config.action_dispatch.default_headers.merge!(
+      'Content-Security-Policy' => "frame-ancestors 'self' https://developers.login.gov/" 
+    )
```

Comment:
I'd worry this overrides the entire CSP, which can have many different directives for many different purposes.

Elsewhere, we'll use Rails helpers to be more precise with which CSP directives are to be modified. Would it be possible to do something similar here?

https://github.com/18F/identity-idp/blob/bc9da12f2cbe63c26410159390ba2d4498e50d0c/app/controllers/concerns/secure_headers_concern.rb#L14-L18

Reply:
makes sense, I'll take a look!

Reply:
@aduth  I moved the logic into that method, let me know what you think